consensus/validator.go

```diff
@@ -272,6 +270,11 @@ func (consensus *Consensus) onCommitted(msg *msg_pb.Message) {
 
 	consensus.FBFTLog.AddMessage(recvMsg)
 
+	if recvMsg.BlockNum > consensus.blockNum {
```

Comment:
If a prepared message is missing, onCommitted will never spin state sync since the it is already returned in 

```
	blockObj := consensus.FBFTLog.GetBlockByHash(recvMsg.BlockHash)
	if blockObj == nil {
		consensus.getLogger().Debug().
			Uint64("blockNum", recvMsg.BlockNum).
			Uint64("viewID", recvMsg.ViewID).
			Str("blockHash", recvMsg.BlockHash.Hex()).
			Msg("[OnCommitted] Failed finding a matching block for committed message")
		return
	}
```

Is this what we expect?

Reply:
yes, we want to spin up the sync when it's very certain. Don't want to turn to sync mode prematurely. It's risky.

Reply:
I mean this sync code is unreachable since the block from prepared message is missing.

Reply:
Ok, then we probably don't need to have this sync code in oncommitted. What do you think?

Reply:
OK, it's actually a trade off between catch up performance and how sync mode effects the rest of validator consensus logic. I definitely agree with reducing complexity of the consensus logic for now, and refactoring the message handling logic for sync mode can be a good todo in the future.